packages/vue/vue-vanilla/src/array/ArrayListRenderer.vue

```diff
@@ -5,7 +5,12 @@
         :class="styles.arrayList.addButton"
         type="button"
         @click="addButtonClick"
-      >
+        :disabled="!control.enabled ||
+            (arraySchema !== undefined &&
+            arraySchema.maxItems !== undefined &&
+            control.data !== undefined &&
+            control.data.length >= arraySchema.maxItems)"
+        >
```

Comment:
I would prefer moving the calculation of this into a `computed`. Also we should only disable the add button if `appliedOptions.restrict` is `true`. By default we don't interfere with the user doing something which invalidates the schema.
The same could be implemented for the delete button and checking the `minItems` if you want to.

Reply:
Thanks @sdirix I've addressed this, plus your other feedback. But I had two comments:

* I did initially add appliedOptions.restrict. But I removed it due to the following
   * I cannot see this check being performed on any other renderers in the project (I do see it in use in the vuetify renderers). So I left it out in order to remain consistent with the other renderers
   * I cannot find documentation on how to set this option when using json-forms in vue. If you could point me in this direction, it would be helpful!
* Secondly, I'm afraid I didn't implement a minItems delete button check. In my use-case (vanilla-vue) I cannot see the delete button render. So perhaps this need to first be addressed as a separate issue. I'll post some screenshots below and maybe you can determine if this is due to my usage, or an issue in the package itself